hypothesis-python/tests/cover/test_database_backend.py

```diff
@@ -160,3 +168,128 @@ def test_multiplexed_dbs_read_and_write_all():
     multi.delete(b"c", b"cc")
     for db in (a, b, multi):
         assert set(db.fetch(b"c")) == set()
+
+
+def test_ga_require_readonly_wrapping():
+    database = GitHubArtifactDatabase("test", "test")
+    # save, move and delete can only be called when wrapped around ReadonlyDatabase
+    with pytest.raises(RuntimeError):
```

Comment:
```suggestion
    with pytest.raises(RuntimeError, match=re.escape(database._read_only_message)):
```
etc below - just nice to make sure we get the _right_ `RuntimeError` when it's this easy.